tensorflow_addons/optimizers/moving_average.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow_addons.utils import keras_utils
+
+
+@keras_utils.register_keras_custom_object
+class MovingAverage(tf.keras.optimizers.Optimizer):
+    """Optimizer that computes a moving average of the variables.
+
+    Empirically it has been found that using the moving average of the trained
+    parameters of a deep network is better than using its trained parameters
+    directly. This optimizer allows you to compute this moving average and swap
+    the variables at save time so that any code outside of the training loop
+    will use by default the average values instead of the original ones.
+
+    Example of usage:
```

Comment:
we might want to put examples like these into colab notebooks and not into the documentation text since the documentation becomes quiet bloated. @facaiy @seanpmorgan I know tensorflow core is doing this in there documentation sometimes. What do you guys think

Reply:
It would make sense if the example was big, but it's too small to warrant a colab notebook. Also, including an example in the method makes sense for anyone looking at the source to figure out how it works, as opposed to looking up a specific colab notebook.

Reply:
I can implement a Colab Notebook along with the docstring. That would be the best of both worlds, IMHO.

Reply:
Thanks for ping me, Moritz, Dheeraj.
I think It's fine to write a small example in the document, which is convenient for both users and developers to understand how to use it quickly. However, it would be better (and appreciated) to add an toy example in colab notebooks.

Reply:
I'll start working on that colab notebook.